### PR TITLE
Support Uint8Array format when importing Solana private key

### PR DESCRIPTION
**Pull request by darkdh**
Resolves https://github.com/brave/brave-browser/issues/26672

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Setup wallet on https://solflare.com/
2. Export private key on Solflare and import it into Brave Wallet through import Solana key
3. Import should be successful and address should be same as Solflare

### CODE REVIEW

File: components/brave_wallet/common/solana_utils_unittest.cc [hunk 1]
```diff
@@ -140,4 +140,46 @@ TEST(SolanaUtilsUnitTest, IsBase58EncodedSolanaPubkey) {
...
+  for (const auto& valid_case : valid_cases) {
+    std::vector<uint8_t> bytes;
+    EXPECT_TRUE(Uint8ArrayDecode(valid_case.input, &bytes, valid_case.len));
+    EXPECT_EQ(bytes, valid_case.output);
+  }
+
+  const struct {
+    const char* input;
+    const size_t len;
+  } invalid_cases[] = {{"[34}", 1},
```
Comment (supermassive) [inline components/brave_wallet/common/solana_utils_unittest.cc:169]: Worth also test for `"[]"`, `"[,]"`, `""`, `"["`, `"]"`

File: components/brave_wallet/common/solana_utils.cc [hunk 2]
```diff
@@ -82,4 +85,27 @@ bool IsBase58EncodedSolanaPubkey(const std::string& key) {
   return Base58Decode(key, &bytes, kSolanaPubkeySize);
 }
 
+bool Uint8ArrayDecode(const std::string& str,
+                      std::vector<uint8_t>* ret,
+                      size_t len) {
+  if (!base::StartsWith(str, "[") && !base::EndsWith(str, "]"))
```
Comment (supermassive) [inline components/brave_wallet/common/solana_utils.cc:91]: As users usually copy-paste this string it makes sense to trim it before checking for []
Actually not sure if trimming already done somewhere by callers.

**Comment by supermassive** [comment]
LGTM

File: components/brave_wallet/common/solana_utils_unittest.cc (hunk 1 above)
Comment (darkdh) [reply components/brave_wallet/common/solana_utils_unittest.cc]: added and squashed thx.

File: components/brave_wallet/common/solana_utils.cc (hunk 2 above)
Comment (darkdh) [reply components/brave_wallet/common/solana_utils.cc]: Good point, trimming added in `KeyringService::ImportAccount`

**Pull request merged**